drivers/flash/Kconfig

```diff
@@ -162,4 +162,15 @@ config SOC_FLASH_MCUX_DEV_NAME
 	help
 	  Specify the device name for the flash driver.
 
+config DRV_SOC_FLASH_WRITE_BLOCK_SIZE
+	int "write block size of the SoC driver (fixed)"
+	depends on FLASH
+	range 1 1 if SOC_FLASH_NRF5
+	range 4 4 if SOC_FLASH_STM32
+	range 4 4 if SOC_FLASH_QMSI
+	range -1 -1
+	help
+	 The minimum write block size supported by the SoC flash driver.
```

Comment:
is this in bytes? (if so make that clear in the comments).  Also remove 'SoC' here.

Reply:
Are you surre? It is possible to have enabled SPI flash driver and SoC flash driver simultaneously.

Reply:
Ok, so another reason to move this to the flash driver api, since we can support multiple different flash drivers.

Reply:
This will make impossible compile time optimization of a module which will use the driver. And there is always only one SoC flash driver.

Reply:
how significant is such optimization really going to be?  I know there was some mention of some filesystem code needing this, can we see what that could that would use this look like.

Seems like we can have systems that have both internal and external flash that would need this option set, and possibly set different for each flash interface/driver.

Reply:
NFFS currently requires 1-byte granular unaligned access to flash. Starting there there is a problem today because we don't have a way of communicating that to NFFS today so it can assert or fail to initialize.